### ISSUE TITLE
Multiple rides appearing on profile

### ISSUE THREAD

**Issue by Goatford** [kinetic-fit/fit-android-issues#32]
Since the latest update, all my current activities have disappeared and multiple other activities  have appeared. 
On 14th April 4 rides appeared
On 15th April 15 rides appeared.
I am now constantly asked to adjust my FTP figure

[Screenshot_20200415-191522_Kinetic](https://user-images.githubusercontent.com/62814498/79372367-bc80fa00-7f4d-11ea-8454-4b7936711cf9.jpg)
![Screenshot_20200415-191529_Kinetic](https://user-images.githubusercontent.com/62814498/79372425-c3a80800-7f4d-11ea-88b0-dabe2fd76d55.jpg)
![Screenshot_20200415-191542_Kinetic](https://user-images.githubusercontent.com/62814498/79372438-c6a2f880-7f4d-11ea-9b6c-bd2aab6f7e46.jpg)
![Screenshot_20200415-191554_Kinetic](https://user-images.githubusercontent.com/62814498/79372457-c86cbc00-7f4d-11ea-996e-7d692601b679.jpg)
![Screenshot_20200415-191604_Kinetic](https://user-images.githubusercontent.com/62814498/79372473-ca367f80-7f4d-11ea-8b88-e01d727938b8.jpg)

**Comment by lexicalninja** [comment]
Please be sure to fill out all parts of a bug report to ensure the fastest solving of issues:
**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6]
 - OS: [e.g. iOS8.1]
 - Kinetic Fit Version [e.g. 2.0.1]

**Trainer (please complete the following information):**
- Trainer: [e.g. SmartControl 1]
- Firmware Version [e.g. v2.1]

**Additional context**
Add any other context about the problem here.

Also, please be specific about what you mean by rides appearing. Are they not your rides? 

When are you being asked to update your FTP? What screen is that happening on?

**Comment by jacco411** [comment]
I'm also having the same problem:

> **Describe the bug**
> When I open the app, different workouts that don't belong to me appear. I also can barely find the works I did.
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Go to the main page of the app
> 2. Click on nothing
> 3. Scroll down to look at recent workouts
> 4. See error - many workouts appear that aren't mine and multiple workouts for each day appear.
> 
> **Expected behavior**
> To only see my own workouts
> 
> **Screenshots**
> 
![Screenshot_20200415-163931_Kinetic](https://user-images.githubusercontent.com/62756796/79386678-7c634c80-7f38-11ea-9334-100cc1913b0d.jpg)

> 
> **Smartphone (please complete the following information):**
> 
> * Device: Samsung Galaxy s9+
> * OS: Android 10
> * Kinetic Fit Version 2.5.5 (236)
> 
> **Trainer (please complete the following information):**
> 
> * Trainer: rock n roll control
> * Firmware Version 1.3
> 
> **Additional context**
> It is not asking me to update my ftp. That is correct. Otherwise, similar issue.

**Comment by lexicalninja** [comment]
send me your username/email for logging in so I can look at what is in your profiles, please

**Comment by Goatford** [comment]
Reproduce
Steps to reproduce the behavior:
Open Kinetic App

A clear and concise description of what you expected - 
To see my activities.

Attachments - 
See above.

Smartphone (please complete the following information):

Device: Samsung S9
OS: Andriod 10
Kinetic Fit Version [e.g. 2.5.5
Trainer (please complete the following information):

Trainer: N/A
Firmware Version N/A
Additional context
Add any other context about the problem here.

Also, please be specific about what you mean by rides appearing. Are they not your rides?
The rides displayed on the attachments are not mine. The rides I completed prior to the latest update have vanished

When are you being asked to update your FTP? What screen is that happening on?
It tells me my FTP figure has changed after the latest uploaded rides and would I like to change it.

**Comment by lexicalninja** [comment]
what screen is asking you to update your ftp?
the analysis screen?
the main page?
another screen?

**Comment by jacco411** [comment]
Lam.larry8@gmail.com

**Comment by lexicalninja** [comment]
thanks. I will look into this and try and figure out what happened. I don't see it in my account here, so it is very odd.

**Comment by lexicalninja** [comment]
this is really crazy.
I went and had my device pull down session history for your account and I got these.
![Screenshot_20200415-161113](https://user-images.githubusercontent.com/7376126/79389223-efb68f80-7f33-11ea-8f69-c0cb1747b9f2.png)

I assume that these are all your workouts for the last few weeks, correct?

If so, please log out and log back in and let me know what happens. If you still see this, please clear the app cache from settings and try again.

**Comment by lexicalninja** [comment]
Also, do you share a device with anyone for kinetic or are you the only ones on your devices?

**Comment by jacco411** [comment]
So, crazy. I logged out, logged back in and now it's normal. It's missing my workout from last night, though. To answer any previous questions, yes, what you pulled up was my workouts and, no, I do not share my device with anyone else.

**Comment by lexicalninja** [comment]
I found the cause of this. It was me. I changed up part of the authentication system and the older versions didnt have a personal key and it was using a default value. 

logging back in saved the key, so that is why that fixed that issue.

the missing workout got saved to a blank account with everyone else with the same issue. I am going to see if there is a way to move all of those back for everyone and let you know.

I am adding the fix to the Monday release, so if anyone new comes to this thread with the same issue, logging out and logging back in will fix the major issue and I am working to get your workout data back to you.
